File: app/src/main/java/tourguide/tourguidedemo/NavDrawerActivity.java
package tourguide.tourguidedemo;

import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.widget.Toolbar;
import android.view.Gravity;
import android.view.View;
import android.view.ViewTreeObserver;
import android.widget.TextView;

import tourguide.tourguide.Overlay;
import tourguide.tourguide.Pointer;
import tourguide.tourguide.Sequence;
import tourguide.tourguide.ToolTip;
import tourguide.tourguide.TourGuide;

public class NavDrawerActivity extends ActionBarActivity {
    TextView mTextView1, mTextView2, mTextView3;
    NavDrawerActivity mActivity;
    Sequence mSequence;
    TourGuide mTutorialHandler;
    DrawerLayout mDrawerLayout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mActivity = this;
        setContentView(R.layout.activity_nav_drawer);

        /* get views from xml */
        mTextView1 = (TextView)findViewById(R.id.item1);
        Toolbar toolbar = (Toolbar)findViewById(R.id.toolbar);
        mDrawerLayout = (DrawerLayout)findViewById(R.id.drawer_layout);

        /* setup toolbar */
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
        toolbar.setTitle("Nav Drawer Example");
        mTutorialHandler = TourGuide.init(mActivity).with(TourGuide.Technique.Click)
                .setPointer(new Pointer())
                .setToolTip(new ToolTip().setTitle(null).setDescription("hello world"))
                .setOverlay(new Overlay().setBackgroundColor(Color.parseColor("#66FF0000")));


        final ActionBarDrawerToggle mDrawerToggle = new ActionBarDrawerToggle(
                this,  mDrawerLayout, toolbar, R.string.drawer_open_string, R.string.drawer_close_string){
            @Override
            public void onDrawerOpened(View drawerView) {
                super.onDrawerOpened(drawerView);
                /* We need call playOn only after the drawer is opened,
                   so that TourGuide knows the updated location of the targetted view */
                mTutorialHandler.playOn(mTextView1);
            }
        };
        mDrawerLayout.setDrawerListener(mDrawerToggle);
        mDrawerToggle.syncState();

        /* setup clean up code */
        mTextView1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mTutorialHandler.cleanUp();
                mDrawerLayout.closeDrawers();
            }
        });

        final ViewTreeObserver viewTreeObserver = mTextView1.getViewTreeObserver();
        viewTreeObserver.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
            @Override
            public void onGlobalLayout() {
                mTextView1.getViewTreeObserver().removeGlobalOnLayoutListener(this);// make sure this only run once
                mDrawerLayout.openDrawer(Gravity.LEFT);
            }
        });
    }

}

File: app/src/main/java/tourguide/tourguidedemo/NavDrawerActivity.kt
package tourguide.tourguidedemo

import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.view.Gravity
import android.view.View
import android.view.ViewTreeObserver
import kotlinx.android.synthetic.main.activity_nav_drawer.*
import tourguide.tourguide.TourGuide

class NavDrawerActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_nav_drawer)

        /* setup toolbar */
        setSupportActionBar(toolbar.apply { title = "Nav Drawer Example" })
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setHomeButtonEnabled(true)
            setDisplayShowTitleEnabled(false)
        }

        val tourGuide = TourGuide.create(this) {
            pointer { }
            overlay { backgroundColor { Color.parseColor("#66FF0000") } }
            toolTip {
                title { "" }
                description { "hello world" }
            }
        }

        val drawerToggle = object : ActionBarDrawerToggle(
                this, drawerLayout, toolbar, R.string.drawer_open_string, R.string.drawer_close_string) {
            override fun onDrawerOpened(drawerView: View?) {
                super.onDrawerOpened(drawerView)
                /* We need call playOn only after the drawer is opened,
                   so that TourGuide knows the updated location of the targetted view */
                tourGuide.playOn(textView1)
            }
        }
        drawerLayout.setDrawerListener(drawerToggle)
        drawerToggle.syncState()

        /* setup clean up code */
        textView1.setOnClickListener {
            tourGuide.cleanUp()
            drawerLayout.closeDrawers()
        }

        textView1.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
            override fun onGlobalLayout() {
                textView1.viewTreeObserver.removeGlobalOnLayoutListener(this)// make sure this only run once
                drawerLayout.openDrawer(Gravity.LEFT)
            }
        })
    }

}
